apps/confessions-client/components/PublishConfession.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  constructPassportPcdGetRequestUrl,
+  usePassportPCD,
+  useSemaphorePassportProof,
+} from "@pcd/passport-interface";
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import { SemaphoreGroupPCDPackage } from "@pcd/semaphore-group-pcd";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { sha256 } from "js-sha256";
+import { PASSPORT_URL, SEMAPHORE_GROUP_URL, requestProofFromPassport } from "../src/util";
+import { postConfession } from "../src/api";
+
+export function PublishConfession({
+  onPublished,
+}: {
+  onPublished: (newConfession: string) => void;
+}) {
+  const [confession, setConfession] = useState<string>("");
+
+  const pcdStr = usePassportPCD()
+  const { proof, valid, error } = useSemaphoreProof(
+    SEMAPHORE_GROUP_URL,
+    confession,
+    pcdStr
+  )
+
+  useEffect(() =>  {
+    if (valid === undefined) return; // verifying
+
+    if (error) {
+      // TODO: display error to the user
+      console.error("error using semaphore passport proof", error);
+      return;
+    }
+
+    if (!valid) {
+      // TODO: display error to the user
+      console.error("proof is invalid");
+      return;
+    }
+
+    (async () => {
+      const res = await postConfession(SEMAPHORE_GROUP_URL, confession, pcdStr);
+      if (!res.ok) {
+        // TODO: display error to the user
+        const err = await res.text();
+        console.error("error posting confession to the server:", err);
+      }
+    })().then(
+      () => {
+        onPublished(confession)
+      }
+    )
+  }, [valid, error, confession, pcdStr, onPublished]);
+
+  return (
+    <>
+      <h2>Publish confession</h2>
+      <BigInput
+        placeholder="Confession"
+        type="text"
+        value={confession}
+        onChange={(e) => setConfession(e.target.value)}
+      />
+      <br />
+      <br />
+      <button
+        onClick={useCallback(
+          () => requestSemaphoreProof(confession),
+          [confession]
+        )}
+      >
+        Publish
+      </button>
+      {proof != null && (
+        <>
+          <h3>Got Zuzalu Member Confession Proof from Passport</h3>
+          <pre>{JSON.stringify(proof, null, 2)}</pre>
+          {valid === undefined && <p>❓ Proof verifying</p>}
+          {valid === false && <p>❌ Proof is invalid</p>}
+          {valid === true && <p>✅ Proof is valid</p>}
+        </>
+      )}
```

Comment:
for the version that we release as an example, we should hide this

Reply:
deleted this
added a console.log(proof), thought might still be useful for developers as this is a demo app